src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
+
+    return next.handle().pipe(
+      map((data) => {
+        const response = {
+          isSuccess: statusCode >= 200 && statusCode < 300 ? true : false,
+          statusCode: statusCode,
+          error: data && data.message ? data.message : '',
+          message: data && data.message ? data.message : '',
+          data: data && data.message ? { ...data, message: undefined } : data,
```

Comment:
PullProofAI

There is redundant code in mapping the response object. The conditions for setting `error`, `message`, and `data` fields are repeated unnecessarily. This can be simplified to improve readability and maintainability.

```diff
-           error: data && data.message ? data.message : '',
-           message: data && data.message ? data.message : '',
-           data: data && data.message ? { ...data, message: undefined } : data,
+           error: data?.message || '',
+           message: data?.message || '',
+           data: data?.message ? { ...data, message: undefined } : data,
```

#### Performance Improvement

The mapping logic can be optimized by reducing unnecessary checks and improving the readability of the code. By using optional chaining (`?.`) and simplifying the conditions, the code becomes more concise and easier to understand.

This updated approach ensures that the `error`, `message`, and `data` fields are set based on the presence of `data.message` without repeating the same condition multiple times.


<!-- This is an auto-generated comment by PullProofAI -->